conda_smithy/templates/azure-pipelines-linux.yml.tmpl

```diff
@@ -5,10 +5,30 @@
 jobs:
 - job: linux
   pool:
+  {%- if azure.pool is defined %}
```

Comment:
I think we should have something like,
```
azure: {
  pool_linux: {
     vmIMage: "ubuntu-16.04"
  }
  pool_osx: {
    vmImage: macOS-10.14
  }
}
```
and then json dump the whole `pool_linux` here.

Reply:
sorry, yaml dump

Reply:
Not sure what's best to do at the yaml level, given that 
```yaml
pool:
  vmImage: some_img
```
and
```yaml
pool:
  name: pool_name

or 

pool: pool_name
```
are mutually exclusive. Any suggestion, @isuruf?

Reply:
I meant you still need a jinja check to overlook `vmImage` if `name` or `pool` is defined.

Reply:
In conda-forge.yml, you can override, the `azure: pool_linux: {vmImage: 'ubuntu-16.04'}` with whatever you need. `azure: pool_linux: {name: Our_Local_Pool, demands: '- tagA -equals 2'}`

Reply:
We use the yaml object `azure.pool_linux` from the config and dump it here.

Reply:
I guess you wanna minimize the logic in the template, @isuruf? I am putting together an alternative. What should I do with `workspace`? It's valid only with self-hosted agents, but it's an attribute of `job`, not `pool`.

Reply:
Ah, let me try something.

Reply:
Maybe we can do,
```yaml
azure:
  yaml_linux:
    pool:
      vmImage: 'ubuntu-16.04'
    strategy:
      maxParallel: 1
````
and override that with,
```yaml
azure:
  yaml_linux:
    pool:
      name: Our_Local_Pool
      demands:
        - tagA -equals 2
    workspace:
      clean: all
    strategy:
      maxParallel: 1
````

Reply:
NVM, `strategy` has a `matrix`.

Reply:
Do you mean to do this as the default for `conda-forge.yml` or  for `_load_forge_config`?

Reply:
How do I dump Python dicts in yaml with the formats
1. key : value
2. - some string

with a simple loop and without `is defined` checks?

Reply:
Let me try to figure it out and get back to you.

Reply:
@isuruf How does this look?
```diff
diff --git a/conda_smithy/templates/azure-pipelines-linux.yml.tmpl b/conda_smithy/templates/azure-pipelines-linux.yml.tmpl
index ae5a7a1..a149423 100644
--- a/conda_smithy/templates/azure-pipelines-linux.yml.tmpl
+++ b/conda_smithy/templates/azure-pipelines-linux.yml.tmpl
@@ -5,27 +5,20 @@
 jobs:
 - job: linux
   pool:
-  {%- if azure.pool is defined %}
-    # self-hosted agent
-    {%- if azure.pool.name is defined %}
-    name: {{ azure.pool.name }}
+  {%- for spec in azure.pool_linux %}
+    {%- if azure.pool_linux[spec] is string %}
+    {{ spec }}: {{ azure.pool_linux[spec] }}
     {%- else %}
-    name: {{ azure.pool }}
-    {%- endif %}
-  {%- if azure.pool.demands is defined %}
-    demands:
-    {%- for requirement in azure.pool.demands %}
-    - {{ requirement }}
+    {{ spec }}:
+    {%- for item in azure.pool_linux[spec] %}
+    - {{ item }}
     {%- endfor %}
-  {%- endif %}
+    {%- endif %}
+  {%- endfor %}
   {%- if azure.workspace is defined %}
   workspace:
     clean: {{ azure.workspace.clean }}
   {%- endif %}
-  {%- else %}
-    # MS-hosted agent
-    vmImage: ubuntu-16.04
-  {%- endif %}
   timeoutInMinutes: {{ azure.timeout_minutes }}
   strategy:
     maxParallel: {{ azure.strategy.maxParallel }}
```